library/std/src/sync/mpmc/mod.rs

```diff
@@ -62,6 +62,9 @@ pub fn channel<T>() -> (Sender<T>, Receiver<T>) {
 ///
 /// A special case is zero-capacity channel, which cannot hold any messages. Instead, send and
 /// receive operations must appear at the same time in order to pair up and pass the message over.
+///
+/// # Panics
+/// Panics if `cap` exceeds [`isize::MAX`] *bytes*.
```

Comment:
I think this would fail even before `isize::MAX` right? Since `Slot` is larger than one byte here https://github.com/rust-lang/rust/blob/4e5a732fbb02ab64a1e8ce2d7abcd64442d275fd/library/std/src/sync/mpmc/array.rs#L104-L109.

Reply:
I'm not sure we want to guarantee a particular panic point -- `Slot` is an internal implementation detail. On 64-bit architectures this seems likely to be more commonly encountered through memory exhaustion, too, which isn't too useful to document.

I'm going to nominate for libs-api to see what we want to document here.

Reply:
Agreed. Maybe we could say something that indicates this a huge `cap` may cause allocation failure, without giving any expectation for what that value is.